server/src/Utopia/Web/Executors/Common.hs

```diff
@@ -431,6 +432,31 @@ getGithubBranch githubResources awsResource logger metrics pool userID owner rep
       pure (projectContent, commitSha)
     pure $ either getBranchContentFailureFromReason getBranchContentSuccessFromContent result
 
+convertUsersRepositoriesResultToUnfold :: Int -> GetUsersPublicRepositoriesResult -> Maybe (GetUsersPublicRepositoriesResult, Maybe Int)
+convertUsersRepositoriesResultToUnfold page result =
+  -- We expect some X elements per page, if we get less than that, assume we've hit the end and should not
+  -- continue paginating.
+  Just (result, if Protolude.length result < userRepositoriesPerPage then Nothing else Just (page + 1))
```

Comment:
Could this logic be extracted and reused (e.g. in the `convertBranchesResultToUnfold` as well)?

Reply:
Quite possibly, but since there's a constant in there I didn't think there was much gain in generalising it given it's basically a one-liner.